binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/function/SerdesProvidedAsBeansTests.java

```diff
@@ -42,78 +43,178 @@
 import org.springframework.kafka.test.EmbeddedKafkaBroker;
 import org.springframework.kafka.test.condition.EmbeddedKafkaCondition;
 import org.springframework.kafka.test.context.EmbeddedKafka;
-import org.springframework.util.Assert;
 
-@EmbeddedKafka
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests to verify proper resolution of {@link Serde serdes} when they are provided as beans.
+ *
+ * @author Soby Chako
+ * @author Chris Bono
+ */
+@EmbeddedKafka(topics = { "topic1", "topic2" })
 public class SerdesProvidedAsBeansTests {
 
 	private static final EmbeddedKafkaBroker embeddedKafka = EmbeddedKafkaCondition.getBroker();
 
 	@Test
-	void testKstreamWordCountFunction() throws NoSuchMethodException {
-		SpringApplication app = new SpringApplication(SerdeProvidedAsBeanApp.class);
+	void simpleSerdeBeansAreResolvedProperly() throws Exception {
```

Comment:
I ended up adding to this related existing test rather than have 1 for generics and 1 for simple types. There is some duplicated code I will refactor when I do that follow on PR I mention in the other comments.

Reply:
To be handled in #2370